README.md

```diff
@@ -1,26 +1,97 @@
-  - [Using kubectl](#using-kubectl-1)
-  - [Using Pulumi](#using-pulumi-1)
-  - [Extended Examples](#extended-examples)
-* [Stack CR Documentation](#stack-cr-documentation)
-* [Prometheus Metrics Integraton](#prometheus-metrics-integration)
-* [Development](#development)
+- [Pulumi Kubernetes Operator](#pulumi-kubernetes-operator)
+  - [Overview](#overview)
+    - [What is Pulumi?](#what-is-pulumi)
+    - [When To Use the Pulumi Kubernetes Operator?](#when-to-use-the-pulumi-kubernetes-operator)
+    - [Prerequisites](#prerequisites)
+      - [Install Pulumi CLI](#install-pulumi-cli)
+      - [Login to Your Chosen State Backend](#login-to-your-chosen-state-backend)
+  - [Deploy the Operator](#deploy-the-operator)
+    - [Using kubectl](#using-kubectl)
+    - [Using Pulumi](#using-pulumi)
+  - [Create Pulumi Stack CustomResources](#create-pulumi-stack-customresources)
+    - [Using kubectl](#using-kubectl-1)
+    - [Using Pulumi](#using-pulumi-1)
+    - [Extended Examples](#extended-examples)
+  - [Stack CR Documentation](#stack-cr-documentation)
+  - [Prometheus Metrics Integration](#prometheus-metrics-integration)
+  - [Development](#development)
+
+### What is Pulumi?
+
+Pulumi is an open source infrastructure-as-code tool for creating, deploying, and managing cloud infrastructure in the programming language of your choice. If you are new to Pulumi, please consider visiting the [getting started](https://www.pulumi.com/docs/get-started/) first to familiarize yourself with Pulumi and concepts such as [Pulumi stacks](https://www.pulumi.com/docs/intro/concepts/stack/) and [backends](https://www.pulumi.com/docs/intro/concepts/state/).
+
+### When To Use the Pulumi Kubernetes Operator?
+
+The Pulumi Kubernetes Operator enables Kubernetes users to create a Pulumi Stack as a first-class Kubernetes API resource, and use the StackController to drive the updates. It allows users to adopt a GitOps workflow for managing their cloud infrastructure using Pulumi. It lives along-side other CI/CD integrations for Pulumi such as with [Github Actions](https://www.pulumi.com/docs/guides/continuous-delivery/github-actions/), [Gitlab CI](https://www.pulumi.com/docs/guides/continuous-delivery/gitlab-ci/), [Jenkins](https://www.pulumi.com/docs/guides/continuous-delivery/jenkins/) etc. See the full list of Pulumi's CI/CD integrations [here](https://www.pulumi.com/docs/guides/continuous-delivery/). Since the Pulumi Kubernetes Operator can be deployed on any Kubernetes cluster, it provides turn-key GitOps functionality for Pulumi users running in self-hosted or restricted settings. Moreover, being an instantiation of the [Kubernetes Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/), it allows the state specified in the Stack Custom Resource and the corresponding Git repository to be asserted despite transient failures lending itself nicely to automation scenarios.
```

Comment:
```suggestion
The Pulumi Kubernetes Operator enables Kubernetes users to create a Pulumi Stack as a first-class Kubernetes API resource, and use the StackController to drive the updates. It allows users to adopt a GitOps workflow for managing their cloud infrastructure using Pulumi. It lives alongside other CI/CD integrations for Pulumi such as with [Github Actions](https://www.pulumi.com/docs/guides/continuous-delivery/github-actions/), [Gitlab CI](https://www.pulumi.com/docs/guides/continuous-delivery/gitlab-ci/), [Jenkins](https://www.pulumi.com/docs/guides/continuous-delivery/jenkins/) etc. See the full list of Pulumi's CI/CD integrations [here](https://www.pulumi.com/docs/guides/continuous-delivery/). Since the Pulumi Kubernetes Operator can be deployed on any Kubernetes cluster, it provides turnkey GitOps functionality for Pulumi users running in self-hosted or restricted settings. The [Kubernetes Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/), lends itself nicely to automation scenarios by driving to the specified state and automatically retrying if transient failures are encountered.
```